sdk/tables/azure-data-tables/README.md

```diff
@@ -25,17 +20,287 @@ best parts of Azure cloud.
+    --expiry <expiry-date>
+
+# Table-level SAS
+az storage table generate-sas \
+    --name <table-name>
+```
+
 ## Key concepts
 
+- **TableServiceClient** - A `TableServiceClient` is a client object that enables you to interact with the Table Service in order to create, list, and delete tables.
+- **TableClient** - A `TableClient` is a client object that enables you to interact with a specific table in order to create, update, list, and delete entities within it.
+- **Table** - A table is a collection of entities. Tables don't enforce a schema on entities, which means a single table can contain entities that have different sets of properties.
+- **Entity** - An entity is a set of properties, similar to a database row. An entity in Azure Storage can be up to 1MB in size. An entity in Azure Cosmos DB can be up to 2MB in size. An entity has a partition key and a row key which together uniquely identify the entity within the table.
+- **Properties** - A property is a name-value pair. Each entity can include up to 252 properties to store data. Each entity also has three system properties that specify a partition key, a row key, and a timestamp.
+- **Partition Key** - An entity's partition key identifies the partition within the table to which the entity belongs. Entities with the same partition key can be queried more quickly, and inserted/updated in atomic operations.
+- **Row Key** - An entity's row key is its unique identifier within a partition.
+
+Common uses of the Table service include:
+
+- Storing TBs of structured data capable of serving web scale applications
+- Storing datasets that don't require complex joins, foreign keys, or stored procedures and can be de-normalized for fast access
+- Quickly querying data using a clustered index
+- Accessing data using the OData protocol
+
 ## Examples
-Use the client library for Tables to:
-- manages tables
-- edit and access table contents
+
+- [Authenticate a client](#authenticate-a-client)
+  - [Authenticate with a connection string](#authenticate-with-a-connection-string)
+  - [Authenticate with a Shared Key](#authenticate-with-a-shared-key)
+  - [Authenticate with a Shared Access Signature (SAS)](#authenticate-with-a-shared-access-signature-sas)
+- [Create, List, and Delete Azure tables](#create-list-and-delete-azure-tables)
+  - [Construct a `TableServiceClient`](#construct-a-tableserviceclient)
+  - [Create a table](#create-a-table)
+  - [List tables](#list-tables)
+  - [Delete a table](#delete-a-table)
+- [Create, List, and Delete table entities](##create-list-and-delete-table-entities)
+  - [Construct a `TableClient`](#construct-a-tableclient)
+  - [Create an entity](#create-an-entity)
+  - [List entities](#list-entities)
+  - [Delete an entity](#delete-an-entity)
+
+### Authenticate a client
+
+#### Authenticate with a connection string
+To use a connection string to authorize your client, call the builder's `connectionString` method with your connection string.
+
+```java
```

Comment:
Use embedme tool to add codesnippets to README. This will inject compilable code into the docs and keeps it updated when APIs change.
https://github.com/Azure/azure-sdk-for-java/wiki/Building#code-snippets-in-readme

Reply:
Working on this now. @srnagar how do I embed code snippets that are just a single line or use example variables (e.g. those that aren't a complete, compilable sample?) For example, the one-line snippet here: https://github.com/Azure/azure-sdk-for-java/blob/681b140cd1fab637aa562b4d0878fe1d2b186f68/sdk/tables/azure-data-tables/README.md#delete-a-table

Reply:
For a single line code, use the same pattern but the start and end range will have the same line number. See storage example [here](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/storage/azure-storage-blob/README.md#create-a-blobclient). You might have to view the raw file to see the embedme tag.

`embedme path/to/ReadmeSamples.java#L10-L10`

Reply:
Nevermind, I solved it with some private fields in the ReadmeSamples class.